aws/cloudformation/serverless/serverless-blog-s3.yml

```diff
@@ -0,0 +1,294 @@
+            Value: !Ref CacheControlMaxAge
+          - Name: BRANCH
+            Value: !Ref GitHubBranch
+          - Name: REPO
+            Value: !Ref GitHubRepo
+          - Name: USER
+            Value: !Ref GitHubUser
+      Name: !Ref GitHubRepo
+      ServiceRole: !GetAtt CodeBuildServiceRole.Arn
+
+#--[ CodePipeline ]-------------------------------------------------------------
+  CodePipeline:
+    Type: AWS::CodePipeline::Pipeline
+    Properties:
+      Name: !Ref GitHubRepo
+      RoleArn: !GetAtt CodePipelineServiceRole.Arn
+      ArtifactStore:
+        Type: S3
+        Location: !Ref BuildBucket
+      Stages:
+        - Name: "Source"
+          Actions:
+            - Name: "Download_Source"
+              ActionTypeId:
+                Category: Source
+                Owner: ThirdParty
+                Version: 1
+                Provider: GitHub
+              Configuration:
+                Owner: !Ref GitHubUser
+                Repo: !Ref GitHubRepo
+                Branch: !Ref GitHubBranch
+                OAuthToken: !Ref GitHubToken
+              OutputArtifacts:
+                - Name: source
+              RunOrder: 1
+        - Name: "Build"
+          Actions:
+            - Name: "Build_and_deploy"
+              ActionTypeId:
+                Category: Build
+                Owner: AWS
+                Version: 1
+                Provider: CodeBuild
+              Configuration:
+                ProjectName: !Ref CodeBuild
+              InputArtifacts:
+                - Name: source
+              OutputArtifacts:
+                - Name: build
+              RunOrder: 1
+
+#--[ Content delivery ]--------------------------------------------------
+  
+  CloudFront:
+    Type: 'AWS::CloudFront::Distribution'
+    Properties:
+      DistributionConfig:
+        Origins:
+        - DomainName: !Select [2, !Split ["/", !GetAtt S3WebsiteBucket.WebsiteURL]]
+          Id: S3Redirection
+          CustomOriginConfig:
+            HTTPPort: '80'
+            HTTPSPort: '443'
+            OriginProtocolPolicy: http-only
+        Enabled: true
+        HttpVersion: 'http2'
+        DefaultRootObject: index.html
+        Aliases: 
+        - !Join ['', ['hugoblog.', !Ref DomainName]]
+        DefaultCacheBehavior:
+          AllowedMethods: ["HEAD", "GET"]
+          Compress: false
+          TargetOriginId: S3Redirection
+          ForwardedValues:
+            QueryString: 'true'
+          ViewerProtocolPolicy: redirect-to-https
+        PriceClass: PriceClass_All
+        ViewerCertificate:
+          AcmCertificateArn: !Ref AcmCertificateArn
+          SslSupportMethod: sni-only
+
+  DNSRecordSet:          
+    Type: "AWS::Route53::RecordSet"
+    Properties: 
+      Comment: Recordset to the cloudfront distribution
+      HostedZoneName: !Join ['',[!Ref DomainName,'.']]
+      Name: !Join ['', ['hugoblog.', !Ref DomainName]]
+      ResourceRecords:
+        - !GetAtt CloudFront.DomainName
+      TTL: '30'
```

Comment:
Nope, you need to use `AliasTarget` property since it is an AWS resource. Like this:
```
  Route53RecordSetIPv4:                                                                                                                                                                                                                                       
    Type: AWS::Route53::RecordSet
    Properties:
      AliasTarget:
        DNSName: !GetAtt CloudfrontDistribution.DomainName
        HostedZoneId: Z2FDTNDATAQYW2
      Comment: !Ref BucketName
      HostedZoneName: !Join ['',[!Ref DomainName,'.']]
      Name: !Join ['', ['hugoblog.', !Ref DomainName]]
      Type: A
```

You may also want to add an IPv6 record, same than the example above but with the `Type: AAAA`.